framework/meta-lib/src/tools/wasm_extractor.rs

```diff
@@ -40,30 +40,33 @@ impl WasmInfo {
     pub fn extract_wasm_info(
         output_wasm_path: &PathBuf,
         extract_imports_enabled: bool,
-        check_ei: &Option<EIVersion>,
-        view_endpoints: Vec<&str>,
-    ) -> WasmInfo {
+        check_ei: Option<&EIVersion>,
+        view_endpoints: &[&str],
+    ) -> Result<WasmInfo, WasmError> {
         let wasm_data = fs::read(output_wasm_path)
-            .expect("error occured while extracting information from .wasm: file not found");
+            .expect("error occurred while extracting information from .wasm: file not found");
 
-        let wasm_info = populate_wasm_info(
+        populate_wasm_info(
             output_wasm_path,
-            wasm_data,
+            &wasm_data,
             extract_imports_enabled,
             check_ei,
             view_endpoints,
-        );
-
-        wasm_info.expect("error occured while extracting information from .wasm file")
+        )
     }
 
-    fn create_call_graph(&mut self, body: FunctionBody) {
+    fn create_call_graph(&mut self, body: FunctionBody) -> Result<(), WasmError> {
         let mut instructions_reader = body
             .get_operators_reader()
             .expect("Failed to get operators reader");
 
         let mut call_functions = HashSet::new();
         while let Ok(op) = instructions_reader.read() {
+            if !is_whitelisted(&op) {
+                let op_str = format!("{:?}", op);
+                return Err(WasmError::ForbiddenOpcode(op_str));
```

Comment:
I think it would be smarter to collect them all, instead of exiting on the first one. Easier to investigate, in case they exist.